pyppeteer/target.py

```diff
@@ -17,33 +19,46 @@
 class Target(object):
     """Browser's target class."""
 
-    def __init__(self, targetInfo: Dict, browserContext: 'BrowserContext',
-                 sessionFactory: Callable[[], Coroutine[Any, Any, CDPSession]],
-                 ignoreHTTPSErrors: bool, defaultViewport: Optional[Dict],
-                 screenshotTaskQueue: List, loop: asyncio.AbstractEventLoop
-                 ) -> None:
+    def __init__(
+            self,
+            targetInfo: Dict,
+            browserContext: 'BrowserContext',
+            sessionFactory: Callable[[], Coroutine[Any, Any, CDPSession]],
+            ignoreHTTPSErrors: bool,
+            defaultViewport: Optional[Dict],
+            screenshotTaskQueue: List,
+            loop: asyncio.AbstractEventLoop
+    ) -> None:
         self._targetInfo = targetInfo
         self._browserContext = browserContext
         self._targetId = targetInfo.get('targetId', '')
         self._sessionFactory = sessionFactory
         self._ignoreHTTPSErrors = ignoreHTTPSErrors
         self._defaultViewport = defaultViewport
         self._screenshotTaskQueue = screenshotTaskQueue
+        self._page = None
+        self._workerPromise = None
         self._loop = loop
-        self._page: Optional[Page] = None
 
         self._initializedPromise = self._loop.create_future()
         self._isClosedPromise = self._loop.create_future()
-        self._isInitialized = (self._targetInfo['type'] != 'page'
-                               or self._targetInfo['url'] != '')
+        self._isInitialized = self._targetInfo['type'] != 'page' \
+                              or self._targetInfo['url'] != ''
         if self._isInitialized:
             self._initializedCallback(True)
 
-    def _initializedCallback(self, bl: bool) -> None:
-        # TODO: this may cause error on page close
-        if self._initializedPromise.done():
-            self._initializedPromise = self._loop.create_future()
-        self._initializedPromise.set_result(bl)
+    def _initializedCallback(self, success: bool) -> None:
+        if not success:
+            return self._initializedPromise.set_result(False)
+        # TODO below seems to always return True - why is even here?
+        opener = self.opener
+        if not opener or not opener._page or self.type != 'page':
+            return self._initializedPromise.set_result(True)
+        openerPage = opener._page
+        if not openerPage.listenerCount(Events.Page.Popup):
+            return self._initializedPromise.set_result(True)
+        openerPage.emit(Events.Page.Popup, openerPage)
+        return self._initializedPromise.set_result(True)
```

Comment:
As far as I can tell, this is what it's trying to do, but apparently puppeteer contributors, really, really love early returns. IMO mine is way more readable :)
```suggestion
        if success:
            opener = self.opener
            if opener and opener._page and self.type == 'page':
                if opener._page.listenerCount(Events.Page.Popup):
                    opener._page.emit(Events.Page.Popup, opener._page)
            return self._initializedPromise.set_result(True)
        return self._initializedPromise.set_result(False)
```

Reply:
I'd keep with early returns which also are more pythonic and easier to keep in line with line length limit.